gk/fib.c

```diff
@@ -1732,6 +1759,136 @@ del_fib_entry(const char *ip_prefix, struct gk_config *gk_conf)
 	return del_fib_entry_numerical(&prefix_info, gk_conf);
 }
 
+/*
+ * Stack when function starts:
+ *
+ * 5 |  gw_addrs  |
+ * 4 |  gt_addrs  |
+ * 3 |   gk_conf  | (unused in this function)
+ * 2 |    table   |
+ * 1 | prefix_str | (unused in this function)
+ *   |____________|
+ */
+static int
+read_grantor_lb_entries(lua_State *l, lua_Integer tbl_size)
+{
+	struct ipaddr *gt_addrs = (struct ipaddr *)lua_touserdata(l, 4);
+	struct ipaddr *gw_addrs = (struct ipaddr *)lua_touserdata(l, 5);
+	lua_Integer i;
+
+	/* Iterate over a table of tables. */
+	for (i = 1; i <= tbl_size; i++) {
+		const char *gt_ip, *gw_ip;
+		int ret;
+
+		/* Get the table at index i. */
+		lua_rawgeti(l, 2, i);
+
+		/*
+		 * Make sure that the item inside
+		 * the table is a table itself.
+		 */
+		if (!lua_istable(l, 6))
+			luaL_error(l, "Grantor entry %ld is not a table\n", i);
+
+		lua_getfield(l, 6, "gt_ip");
+		lua_getfield(l, 6, "gw_ip");
+
+		gt_ip = luaL_checkstring(l, 7);
+		gw_ip = luaL_checkstring(l, 8);
+
+		ret = convert_str_to_ip(gt_ip, &gt_addrs[i - 1]);
+		if (ret < 0) {
+			luaL_error(l,
+				"Cannot convert Grantor IP %s to bytes\n",
+				gt_ip);
+		}
+
+		ret = convert_str_to_ip(gw_ip, &gw_addrs[i - 1]);
+		if (ret < 0) {
+			luaL_error(
+				l,
+				"Cannot convert gateway IP %s to bytes\n",
+				gw_ip);
+		}
+
+		/* Pop this Grantor/gateway IP pair and strings off stack. */
```

Comment:
Shouldn't hte comment read `/* Pop this Grantor/gateway IP pair and their table off stack. */` ?